cvnp/cvnp.cpp

```diff
@@ -9,7 +9,59 @@ namespace cvnp
     namespace detail
     {
         namespace py = pybind11;
-        
+
+        // Translated from cv2_numpy.cpp in OpenCV source code
+        class CvnpAllocator : public cv::MatAllocator {
+        public:
+            CvnpAllocator() { stdAllocator = cv::Mat::getStdAllocator(); }
+            ~CvnpAllocator() {}
+
+            cv::UMatData* allocate(pybind11::array& a) const {
+                //int dims, const int* sizes, int type, size_t* step) const {
```

Comment:
Since this version of `allocate` is actually not an override of any virtual function of `MatAllocator`, we should perhaps name it otherwise, such as `allocate_link_array`

Reply:
This code was copied directly from OpenCV so I kept their original naming, but you're right that it would be better to have a different name.